rpc/eth/api.go

```diff
@@ -648,33 +644,45 @@ func (api *PublicAPI) GetTransactionReceipt(txHash common.Hash) (map[string]inte
 	return receipt, nil
 }
 
-func (api *PublicAPI) GetLogs(filter filters.FilterCriteria) ([]*ethtypes.Log, error) {
-	api.Logger.Debug("eth_getLogs", "filter", filter)
-
-	startRoundInclusive := client.RoundLatest
-	endRoundInclusive := client.RoundLatest
+// getStartEndRounds is a helper for fetching start and end rounds parameters.
+func (api *PublicAPI) getStartEndRounds(filter filters.FilterCriteria) (uint64, uint64, error) {
+	start := client.RoundLatest
+	end := client.RoundLatest
 	if filter.BlockHash != nil {
 		round, err := api.backend.QueryBlockRound(*filter.BlockHash)
 		if err != nil {
-			return nil, fmt.Errorf("query block round: %w", err)
+			return 0, 0, fmt.Errorf("query block round: %w", err)
 		}
-		startRoundInclusive = round
-		endRoundInclusive = round
-	} else {
-		if filter.FromBlock != nil {
-			round, err := api.roundParamFromBlockNum(ethrpc.BlockNumber(filter.FromBlock.Int64()))
-			if err != nil {
-				return nil, fmt.Errorf("convert from block number to round: %w", err)
-			}
-			startRoundInclusive = round
+		start = round
```

Comment:
```suggestion
		return round, round, nil
```

And move start/end declarations bellow to line 661.